posts/2021/08/github-actions-benchmarks.md

```diff
@@ -0,0 +1,336 @@
+GitHub Actions offers different kinds of events that will trigger a worklow. By default the
+`on.pull_request` event is configured to act on three triggers: `[opened, reopened, synchronize]`.
+In practice, this means every push or after closing+opening the PR. However, there are
+[more triggers](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request)!
+
+Of all the choices, we settled for `labeled`. This means that the workflow will be triggered whenever
+the PR is tagged with a label. To specify which label(s) are able to trigger the workflow, you can use
+an `if` close at the `job` level, [like this](https://github.com/scikit-image/scikit-image/blob/2fa66e5/.github/workflows/benchmarks.yml#L10):
+
+```yaml
+name: Benchmark
+
+on:
+  pull_request:
+    types: [labeled]
+
+jobs:
+  benchmark:
+    if: ${{ github.event.label.name == 'run-benchmark' && github.event_name == 'pull_request' }}
+    name: Linux
+    runs-on: ubuntu-20.04
+    ...
+```
+
+In this example, the PR will only get triggered if the label is named `run-benchmark`. This works surprisingly
+well as a manual trigger! It is also restricted to authorized users (triaging permissions or above), so no
+need to fiddle with authentication tokens or similar complications.
+
+There is one gotcha, though: the checks UI will be appended to the last push event panel, which can be confusing
+if more commits have been added to the PR since you added the label. Ideally, the checks UI panel would be
+added next to the _@user added the run-benchmark label_ message. Maybe in a
+[future update](https://twitter.com/jaime_rgp/status/1412419232340627467)?
+
+
+# TLDR
+
+We have seen that GitHub Actions is indeed good enough to perform relative benchmarking in a reliable way
+as long as several passes are averaged together (default behaviour in `asv continuous`). This takes a bit
+more time but you can speed it up a bit with `mamba` and `ccache` for compiled libraries. Even in that case,
+it is probably overkill to run it for every push event, so we are using the `pull_request.labeled` trigger
+to let the maintainers decide when to do it on demand.
+
+---
+
+# Useful references
+
+* [scikit-image/scikit-image #5424](https://github.com/scikit-image/scikit-image/pull/5424): The PR where
+  all this was implemented.
+* [asv.readthedocs.io](https://asv.readthedocs.io): The official documentation for `asv`.
+* [Building an Open Source, Continuous Benchmark System](https://wolfv.medium.com/building-an-open-source-continuous-benchmark-system-717839093962)
+* [Conbench: Language-independent Continuous Benchmarking Tool, by Ursa Labs](https://ursalabs.org/blog/announcing-conbench/)
+* [PyPy Speed](https://speed.pypy.org/)
+* [Pandas Speed](http://pandas.pydata.org/speed/pandas/)
+
+# Acknowledgements
+
+Thanks Gregory Lee, Stefan van der Walt and Juan Nunez-Iglesias for their enthusiastic and useful feedback!
+The plots look that pretty thanks to comments provided by John Lee.
```

Comment:
please add a funding acknowledgement similar to:

This work was funded by the Chan-Zuckerberg Institute (CZI) as part of an [Essential Open Source Software for Science grant](https://chanzuckerberg.com/eoss/proposals/gpu-acceleration-rapid-releases-and-biomedical-examples-for-scikit-image/).